velox/expression/CastExpr.cpp

```diff
@@ -100,16 +121,21 @@ VectorPtr CastExpr::castToDate(
   switch (fromType->kind()) {
     case TypeKind::VARCHAR: {
       auto* inputVector = input.as<SimpleVector<StringView>>();
-      applyToSelectedNoThrowLocal(context, rows, castResult, [&](int row) {
-        try {
-          auto inputString = inputVector->valueAt(row);
-          resultFlatVector->set(row, DATE()->toDays(inputString));
-        } catch (const VeloxUserError& ue) {
-          VELOX_USER_FAIL(
-              makeErrorMessage(input, row, DATE()) + " " + ue.message());
-        } catch (const std::exception& e) {
-          VELOX_USER_FAIL(
-              makeErrorMessage(input, row, DATE()) + " " + e.what());
+      const auto& queryConfig = context.execCtx()->queryCtx()->queryConfig();
+      auto isStandardCast = queryConfig.isIso8601();
+      context.applyToSelectedNoThrow(rows, [&](vector_size_t row) {
```

Comment:
Should we use `applyToSelectedNoThrowLocal`? If castFromDateString throws exception, the setNullInResultAtError flag will be ignored.